ArticleTemplates/assets/js/bootstraps/atoms.js

```diff
@@ -79,4 +82,45 @@ function initExpandables(atom) {
     });
 }
 
+function scrollListener() {
+    const containers = [...document.querySelectorAll('figure[data-atom-type="interactive"]')].flatMap(element => element instanceof HTMLElement ? [element] : []);
+    containers.forEach(function (container) {
+        const rect = container.getBoundingClientRect();
+        const iframe = container.querySelector('iframe');
+        let scroll = -rect.top;
+        if (rect.top > 0) {
+            scroll = 0;
+        } else if (rect.top < -rect.height) {
+            scroll = rect.height;
+        }
+        iframe?.contentWindow?.postMessage({
+            kind: 'interactive:scroll',
+            scroll
```

Comment:
Where will this message be handled?  Is it in the interactive atom?
Will it trigger the handler in line 114 below?

Reply:
It will be handled inside the `iframe`, [by the interactive atom](https://github.com/guardian/interactive-generic-ai2html-scrolly/pull/6/files#diff-209ac0f7130fd8396a870601ee59b5423e341e90de81626d0ca553013ac0e884R71-R91).

The `interactive:scroll` handler in this file is in response to the atom’s own messages.